src/main/java/com/refinedmods/refinedstorage/apiimpl/storage/disk/StorageDiskManager.java

```diff
@@ -90,31 +115,68 @@ public void remove(UUID id) {
 
     @Override
     public void markForSaving() {
-        markDirty();
+        dirty = true;
     }
 
     @Override
-    public void read(CompoundNBT tag) {
-        if (tag.contains(NBT_DISKS)) {
-            ListNBT disksTag = tag.getList(NBT_DISKS, Constants.NBT.TAG_COMPOUND);
-
-            for (int i = 0; i < disksTag.size(); ++i) {
-                CompoundNBT diskTag = disksTag.getCompound(i);
+    public void read() {
+        CompoundNBT tag = readTagFromFile();
+        if (tag.contains(NBT_MINECRAFT_DATA)) {
```

Comment:
Might want to do a warn if the file is corrupt. (e.g. doesn't contain necessary tags)

Reply:
Any corruption should be caught before this by an IO exeption in `readTagFromFile`.  This would also get called whenever it is read before it's generated.